src/main/java/com/faforever/api/data/domain/Login.java

```diff
@@ -53,8 +58,23 @@ public String getUserAgent() {
     return userAgent;
   }
 
-  @OneToOne(mappedBy = "player", fetch = FetchType.LAZY)
-  public BanInfo getBanInfo() {
-    return banInfo;
+  @OneToMany(mappedBy = "player", fetch = FetchType.EAGER)
+  // Permission is managed by BanInfo class
+  @UpdatePermission(expression = "Prefab.Role.All")
+  public List<BanInfo> getBans() {
+    if (this.bans == null) {
+      this.bans = Collections.emptyList();
+    }
+    return this.bans;
+  }
+
+  @Transient
+  public List<BanInfo> getActiveBans() {
+    return getBans().stream().filter(ban -> ban.getBanStatus() == BanStatus.BANNED).collect(Collectors.toList());
+  }
+
+  @Transient
+  public boolean isGlobalBanned() {
+    return getActiveBans().stream().anyMatch(ban -> ban.getLevel() == BanLevel.GLOBAL);
   }
```

Comment:
only internal used to check if an account is locked  and otherwise it will join multiple tables ...